x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared/data_tier_allocation_field.tsx

```diff
@@ -55,14 +55,18 @@ export const DataTierAllocationField: FunctionComponent<Props> = ({
   return (
     <NodesDataProvider>
       {({ nodesByRoles, nodesByAttributes, isUsingDeprecatedDataRoleConfig }) => {
+        const hasDataNodeRoles = Object.keys(nodesByRoles).some((nodeRole) =>
+          nodeRole.trim().startsWith('data_')
```

Comment:
I can imagine someone reading this and being unsure if it's intentional that this evaluates to true for `data_content`, too. How about we make the intention more explicit by using a [type guard](https://stackoverflow.com/questions/50085494/how-to-check-if-a-given-value-is-in-a-union-type-array)?

Building off my other suggestion, `types/index.ts` would become this:

```ts
/**
 * These roles reflect how nodes are stratified into different data tiers.
 *
 */
export type DataTierNodeRole = 'data_hot' | 'data_warm' | 'data_cold';

/**
 * The "data_content" role can store all data the ES stack uses for feature functionality
 * like security related indices.
 */
export type DataNodeRole = 'data_content' | DataTierNodeRole;

/**
 * The "data" role can store data on any tier.
 */
export type AnyNodeRole = 'data' | DataNodeRole;

export const isDataNodeRole = (nodeRole: string): value is DataNodeRole => {
  const dataNodeRoles: string[] = ['data_hot', 'data_warm', 'data_cold', 'data_content'];
  return dataNodeRoles.includes(nodeRole);
};
```

And then this becomes:

```ts
const hasDataNodeRoles = Object.keys(nodesByRoles).some(isDataNodeRole);
```

Reply:
I like the refactor suggestion regarding the data role types! Have implemented a refactor.

I'm not sure I share the same concern regarding intent, but I added a comment for clarity to the existing `startsWith` code :+1:

Reply:
The reason I mentioned it was because I was personally unclear on whether that was the intention, until I read the PR description. :) Thanks for adding the comment.